posts/why-is-react-native-so-promising.md

```diff
@@ -0,0 +1,161 @@
+
+## React Native is a lot like React.
+
+As mentioned previously, in terms of "developer look and feel" React Native and react are very similar. You use classes, ES6, states and the like, with a few notable differences:
+* The atomic JSX elements won't be the same: No HTML tags here, you'll have to use placeholders. `<View>` is the new `<div>`, `<Text>` is the new `<p>`, `<Image>` is the new `<img>`... and then you have other specific components to create buttons, lists, etc. 
+* Because it's a native application you have to manage the navigation. Basic links won't do. The [documentation](https://facebook.github.io/react-native/docs/navigation.html) explains it very well.
+
+### Redux works with React Native
+
+Not much to say here except that Redux, the React developers' favourite state management library, is also working with React Native with absolutely no difference in the way you'll dev.
+    
+### You can do stateless components and use HOC in React Native
+
+Stateless components and High Order Components (HOC) are, for me, the revelation of the year. It allows a greater flexibility in your component creation and is way lighter than the use of React Component class inheritance. Coming from a backend world, I started React by using classes because that's what I'm used to (Java, guys, Java) and I didn't knew much about functional programming. It took a senior front-end dev to open my eyes on this wonder.
+
+But  sometimes new languages yield new philosophies. And in this new era of webcomponents, HOCs are the next best thing. I'll try to explain the core principles while staying as concise as possible.
+
+#### HOC in a few words
+
+A Higher Order Component is essentially composed of one React component and one or many Higher Order Functions. A Higher Order Function is a function that accepts a function as a parameter and returns a React Component. In practice it will act as a component wrapper, you'll be able to add unrelated props or new mechanics; enriching the React component in any way you want. It is particularly interesting to hide logic, complexity or a concern that the Component shouldn't care about. 
+
+The most known Higher Order Function is the Redux `connect` function that allows you to bind the props of a "dumb" Component to the Redux Store. Here's a quick reminder of the function's profile:
+
+```
```

Comment:
add the language on code snippets: `javascript`

Reply:
🚵